test/Microsoft.NET.Sdk.Razor.Tests/StaticWebAssets/UpdateExternallyDefinedStaticWebAssetsTest.cs

```diff
@@ -0,0 +1,255 @@
+        var errorMessages = new List<string>();
+        var buildEngine = new Mock<IBuildEngine>();
+        buildEngine.Setup(e => e.LogErrorEvent(It.IsAny<BuildErrorEventArgs>()))
+            .Callback<BuildErrorEventArgs>(args => errorMessages.Add(args.Message));
+
+        Directory.CreateDirectory(Path.Combine(AppContext.BaseDirectory, "dist", "assets"));
+        File.WriteAllText(Path.Combine(AppContext.BaseDirectory, "dist", "assets", "index-C5tBAdQX.css"), "body { color: red; }");
+        File.WriteAllText(Path.Combine(AppContext.BaseDirectory, "dist", "index.html"), "<html><body></body></html>");
+        var assets = new ITaskItem[] {
+            new TaskItem(
+                Path.Combine(AppContext.BaseDirectory, "dist", "assets", "index-C5tBAdQX.css"),
+                new Dictionary<string, string>
+                {
+                    ["RelativePath"] = "assets/index-C5tBAdQX.css",
+                    ["BasePath"] = "",
+                    ["AssetMode"] = "All",
+                    ["AssetKind"] = "Publish",
+                    ["SourceId"] = "MyProject",
+                    ["CopyToOutputDirectory"] = "PreserveNewest",
+                    ["RelatedAsset"] = "",
+                    ["ContentRoot"] = Path.Combine(AppContext.BaseDirectory, "dist"),
+                    ["SourceType"] = "Discovered",
+                    ["AssetRole"] = "Primary",
+                    ["AssetTraitValue"] = "",
+                    ["AssetTraitName"] = "",
+                    ["OriginalItemSpec"] = Path.Combine(AppContext.BaseDirectory, "dist", "assets", "index-C5tBAdQX.css"),
+                    ["CopyToPublishDirectory"] = "PreserveNewest"
+                }),
+            new TaskItem(
+                Path.Combine(AppContext.BaseDirectory, "dist", "index.html"),
+                new Dictionary<string, string>
+                {
+                    ["RelativePath"] = "index.html",
+                    ["BasePath"] = "",
+                    ["AssetMode"] = "All",
+                    ["AssetKind"] = "Publish",
+                    ["SourceId"] = "MyProject",
+                    ["CopyToOutputDirectory"] = "PreserveNewest",
+                    ["RelatedAsset"] = "",
+                    ["ContentRoot"] = Path.Combine(AppContext.BaseDirectory, "dist"),
+                    ["SourceType"] = "Discovered",
+                    ["AssetRole"] = "Primary",
+                    ["AssetTraitValue"] = "",
+                    ["AssetTraitName"] = "",
+                    ["OriginalItemSpec"] = Path.Combine(AppContext.BaseDirectory, "dist", "index.html"),
+                    ["CopyToPublishDirectory"] = "PreserveNewest"
+                })
+        };
+
+        var fingerprintExpressions = new TaskItem[]
+        {
+            new TaskItem(
+                "React",
+                new Dictionary<string, string>
+                {
+                    ["Pattern"] = "assets/.*-(?<fingerprint>.+)\\..*",
+                })
+        };
```

Comment:
This was taken from the sample app that was producing the error. With the fingerprint inference we can let users define when a file has been fingerprinted via a regex and how to get it. 

This pattern is something we can add to the SPA templates so that users don't have to, and the benefit is that we can mark the files as immutable, which means they won't be requested in successive visits.